### ISSUE TITLE
Breakpoints broken when working with rr

### ISSUE THREAD

**Issue by ymarkovitch** [cgdb/cgdb#266]
Since the new-ui introduction breakpoints don't work with Mozilla rr (https://rr-project.org/).
With the latest git version (08ed6bc), when using  rr, e.g.:

`rr foo`
`rr replay -d cgdb foo`

hitting space or 't' in cgdb mode does nothing: no line marks in the source window, no breakpoints set in gdb. break/delete/disable/clear breakpoints in gdb do not show/erase marks in the source window.

_Sometimes_ the _first_ breakpoint works (i.e. cgdb shows the _first_ breakpoint, after which stops working as described above).

I've attached the logs of the session where I:
 * break main
 * continue
 * attempt to set a breakpoint to line 6 of foo.c by pressing <space> in the source window

rr version 5.4.0
gdb version 10.1

Note that CGDB revision 00e26a33 works OK.

[cgdb_gdb_console_io_log1.txt](https://github.com/cgdb/cgdb/files/6722467/cgdb_gdb_console_io_log1.txt)
[cgdb_gdb_mi_io_log1.txt](https://github.com/cgdb/cgdb/files/6722468/cgdb_gdb_mi_io_log1.txt)
[cgdb_log1.txt](https://github.com/cgdb/cgdb/files/6722469/cgdb_log1.txt)
[foo.zip](https://github.com/cgdb/cgdb/files/6722472/foo.zip)

(and, yes, CGDB is **_magnificent_**, really!)

**Comment by meator** [comment]
Weird. I have tried and I cannot reproduce your issue.

And you should call `rr replay` with `-q` if you don't want garbage in cgdb.

**Comment by ymarkovitch** [comment]
Yes, I know about `rr replay -q`: I just practically never use rr from cgdb directly, only through `rr replay -d cgdb -s 33333` running then cgdb from separate TTY. 

Unfortunately, this issue is quite stably reproduced while debugging real binaries, especially in the most common scenario:
 * Run the binary under rr until the segfault
 * Replay the binary under CGDB until the segfault
 * Several `up`s until interesting point in the stack
 * Attempt to set a breakpoint for `reverse-continue`
 
And, yes, on very simple/small binaries this issue is intermittent (maybe, the size of the displayed source plays role?)

I used CGDB+rr for several years and it always worked rock-stable; more to that, 00e26a3 _still_ works perfectly, so the issue is definitely between 00e26a3 and 08ed6bc.

**Comment by brasko** [comment]
Thank you for reporting this. I'll see what I can discover in the next week.

**Comment by brasko** [comment]
Curious, do you have reproducible steps with a basic example?

**Comment by brasko** [comment]
Ping on this.

**Comment by brasko** [comment]
I just installed rr and did the following:
$ gcc -g -o foo foo.c
$ rr foo

rr: Warning: Your CPU frequency governor is 'powersave'.  rr strongly
    recommends that you use the 'performance' governor.  Not using the
    'performance' governor can cause rr to be at least 2x slower
    on laptops.

    On Fedora-based systems, you can enable the 'performance' governor
    by running the following commands:

    $ sudo dnf install kernel-tools
    $ sudo cpupower frequency-set -g performance

rr: Saving the execution of `foo' to trace directory `/home/bob/.local/share/rr/foo-3'.
$ rr replay -d ../cgdb/cgdb /home/bob/.local/share/rr/foo-3

rr: Warning: Your CPU frequency governor is 'powersave'.  rr strongly
    recommends that you use the 'performance' governor.  Not using the
    'performance' governor can cause rr to be at least 2x slower
    on laptops.

    On Fedora-based systems, you can enable the 'performance' governor
    by running the following commands:

    $ sudo dnf install kernel-tools
    $ sudo cpupower frequency-set -g performance

At this point cgdb, pops up, I type 'b main', 'continue'.
Then I presented with the cgdb window and the breakpoints appear to work fine.

I'm not totally sure how to reproduce this issue, but I'd like to. Any tips?

**Comment by brasko** [comment]
This could be a duplicate of #260
